python/cuml/benchmark/algorithms.py

```diff
@@ -82,48 +83,65 @@ def __init__(
             self.name = cuml_class.__name__
         self.accepts_labels = accepts_labels
         self.bench_func = bench_func
+        self.setup_cpu_func = setup_cpu_func
+        self.setup_cuml_func = setup_cuml_func
         self.cpu_class = cpu_class
         self.cuml_class = cuml_class
         self.shared_args = shared_args
         self.cuml_args = cuml_args
         self.cpu_args = cpu_args
         self.data_prep_hook = data_prep_hook
         self.accuracy_function = accuracy_function
+        self.cpu_obj = None 
+        self.cuml_obj = None 
 
     def __str__(self):
         return "AlgoPair:%s" % (self.name)
 
-    def run_cpu(self, data, **override_args):
+    def run_cpu(self, data):
         """Runs the cpu-based algorithm's fit method on specified data"""
-        if self.cpu_class is None:
-            raise ValueError("No CPU implementation for %s" % self.name)
-        all_args = {**self.shared_args, **self.cpu_args}
-        all_args = {**all_args, **override_args}
+        if self.cpu_obj is None:
+            raise ValueError("No CPU object for %s" % self.name)
 
-        cpu_obj = self.cpu_class(**all_args)
         if self.data_prep_hook:
             data = self.data_prep_hook(data)
         if self.accepts_labels:
-            self.bench_func(cpu_obj, data[0], data[1])
+            self.bench_func(self.cpu_obj, data[0], data[1])
         else:
-            self.bench_func(cpu_obj, data[0])
+            self.bench_func(self.cpu_obj, data[0])
 
-        return cpu_obj
+        return self.cpu_obj
 
-    def run_cuml(self, data, **override_args):
+    def run_cuml(self, data):
         """Runs the cuml-based algorithm's fit method on specified data"""
-        all_args = {**self.shared_args, **self.cuml_args}
-        all_args = {**all_args, **override_args}
 
-        cuml_obj = self.cuml_class(**all_args)
         if self.data_prep_hook:
             data = self.data_prep_hook(data)
```

Comment:
I would see the setup hooks running here and returning a local, much like data prep hook

Reply:
The reason I put `set_up_hook` outside run_cpu`/`run_cuml` functions is I don't want that to be timed. In the case of FIL, we need to use the setup hook for training the xgboost model. If we need to move the setup hook here can we move the timer inside the `run_cpu`/`run_cuml` functions as well?